src/sentry/api/serializers/rest_framework/notification_action.py

```diff
@@ -43,16 +43,48 @@ class NotificationActionSerializer(CamelSnakeModelSerializer):
     Django Rest Framework serializer for incoming NotificationAction API payloads
     """
 
-    integration_id = serializers.IntegerField(required=False)
-    sentry_app_id = serializers.IntegerField(required=False)
-    projects = serializers.ListField(child=ProjectField(scope="project:read"), required=False)
-
-    service_type = serializers.CharField()
-    target_type = serializers.CharField()
-    trigger_type = serializers.CharField()
-
-    target_identifier = serializers.CharField(required=False)
-    target_display = serializers.CharField(required=False)
+    # Optional and not needed for spike protection so not documenting
+    integration_id = serializers.IntegerField(
```

Comment:
this is required for Slack or Pagerduty notification actions

Reply:
i'm unsure of opsgenie because i don't see an opsgenie validation function in https://github.com/getsentry/sentry/blob/master/src/sentry/api/serializers/rest_framework/notification_action.py but it's likely opsgenie requires it too

Reply:
how would a user know the integration id?

Reply:
you can see it in `OrganizationIntegrationSerializer`or `IntegrationSerializer`